web/src/engine/osk/src/input/gestures/browser/keytip.ts

```diff
@@ -91,6 +109,16 @@ export default class KeyTip implements KeyTipInterface {
       let canvasHeight = Math.ceil(2.3 * xHeight) + (ySubPixelPadding); //
 
       kts.top = 'auto';
+
+      if(orientation == 'down') {
+        y += canvasHeight - xHeight;
+        this.tip.style.bottom = '0px';
+        this.tip.style.top = 'unset';
+      } else {
+        this.tip.style.bottom = '';
+        this.tip.style.top = ''; // set by CSS
+      }
```

Comment:
Could we do this styling with a classname instead?

Reply:
Existing element class-name for this keytip-component: `kmw-keytip-tip`

https://github.com/keymanapp/keyman/blob/e4e1345a912b3643178aa7084a095d52658c36a9/web/src/resources/osk/kmwosk.css#L412-L421

https://github.com/keymanapp/keyman/blob/e4e1345a912b3643178aa7084a095d52658c36a9/web/src/resources/osk/kmwosk.css#L452-L458

(Wait, that one _doesn't_ set `top`?)

There are currently no differentiated forms for this element.  Noting that whatever CSS nomenclature gets used is likely to need permanent support... any preferences on class names/patterns to use for handling this?

Perhaps something like...

```css
div.kmw-keytip-tip.up { /* */ }

div.kmw-keytip-tip.down { /* */ }
```

would be satisfactory?

Reply:
A benefit to a simple add-on class like that:  we could use the same text (`up`, `down`) as `KeyTipOrientation` supports as the class names.

Reply:
Doesn't fully answer the question.  So, are you implying that `kmw-top` and `kmw-bottom` would be perfectly satisfactory, with the `kmw-` bit being the only issue?

Reply:
Yes